spec/rspec/support/spec/shell_out_spec.rb

```diff
@@ -33,16 +33,26 @@
   it 'passes along the provided ruby flags' do
     out, err, status = run_ruby_with_current_load_path('puts "version"', '-v')
     expect(out).to include('version', RUBY_DESCRIPTION)
-    expect(err).to eq('')
+    expect(ignore_bundler_warnings err).to eq('')
     expect(status.exitstatus).to eq(0)
   end
 
   it 'filters out the annoying output issued by `ruby -w` when the GC ENV vars are set' do
     with_env 'RUBY_GC_HEAP_FREE_SLOTS' => '10001', 'RUBY_GC_MALLOC_LIMIT' => '16777217', 'RUBY_FREE_MIN' => '10001' do
       out, err, status = run_ruby_with_current_load_path('', '-w')
       expect(out).to eq('')
-      expect(err).to eq('')
+      expect(ignore_bundler_warnings err).to eq('')
       expect(status.exitstatus).to eq(0)
     end
   end
+
+  def ignore_bundler_warnings(input)
+    input.split("\n").reject do |l|
+      # Ignore bundler warning.
+      l =~ %r{bundler/(source/)?rubygems} ||
+      # Ignore bundler + rubygems warning.
+      l =~ %r{site_ruby/\d\.\d\.\d/rubygems} ||
+      l =~ %r{lib/bundler/rubygems}
+    end.join("\n")
+  end
```

Comment:
There's a lot of duplication between this and the other file.  Can this method be pulled into a helper module?